tests/rptest/tests/partition_move_interruption_test.py

```diff
@@ -46,13 +46,53 @@ def __init__(self, ctx, *args, **kwargs):
         self.partition_count = 20
         self.consumer_timeout_seconds = 90
 
+    def get_moving_to_partitions_metrics(self, node):
+        metrics = self.redpanda.metrics_sample("moving_to_partitions", [node])
+        return int(sum([metric.value for metric in metrics.samples]))
+
+    def get_moving_from_partitions_metrics(self, node):
+        metrics = self.redpanda.metrics_sample("moving_from_partitions",
+                                               [node])
+        return int(sum([metric.value for metric in metrics.samples]))
+
+    def metrics_correct(self, prev_assignment, assignmets):
+        prev_assignment_nodes = [a["node_id"] for a in prev_assignment]
+        assignment_nodes = [a["node_id"] for a in assignmets]
+        moving_from_sum = 0
+        moving_to_sum = 0
+        result = True
+        for node in self.redpanda.nodes:
+            node_id = self.redpanda.idx(node)
+            moving_to_partitions_amount = self.get_moving_to_partitions_metrics(
+                node)
+            moving_to_sum += moving_to_partitions_amount
+            moving_from_partitions_amount = self.get_moving_from_partitions_metrics(
+                node)
+            moving_from_sum += moving_from_partitions_amount
+
+            if node_id in prev_assignment_nodes and node_id not in assignment_nodes:
+                result = result and moving_to_partitions_amount == 0 and moving_from_partitions_amount == 1
+            elif node_id not in prev_assignment_nodes and node_id in assignment_nodes:
+                result = result and moving_to_partitions_amount == 1 and moving_from_partitions_amount == 0
+            else:
+                result = result and moving_to_partitions_amount == 0 and moving_from_partitions_amount == 0
+
+        result = result and moving_to_sum == moving_from_sum
+        return result
+
+    def check_metrics(self, prev_assignment, assignmets):
+        return wait_until(
+            lambda: self.metrics_correct(prev_assignment, assignmets),
+            timeout_sec=10)
+
     def _random_move_and_cancel(self, unclean_abort):
         metadata = self.client().describe_topics()
         topic, partition = self._random_partition(metadata)
-        prev_assignment, _ = self._dispatch_random_partition_move(
+        prev_assignment, assignments = self._dispatch_random_partition_move(
             topic=topic, partition=partition, allow_no_op=False)
 
         self._wait_for_move_in_progress(topic, partition)
+        self.check_metrics(prev_assignment, assignments)
         self._request_move_cancel(unclean_abort=unclean_abort,
```

Comment:
should we also check after the cancellation?

Reply:
done